### ISSUE TITLE
Github CI Fequently failing with: "*TaskFactory" could not be loaded

### ISSUE THREAD

**Issue by benaadams** [dotnet/coreclr#6746]
"The task factory "*TaskFactory" could not be loaded from the assembly"

Doesn't seem to be current open issue for it (previous issue https://github.com/dotnet/coreclr/issues/1652)

Upstream issue https://github.com/Microsoft/msbuild/issues/899

@weshaggard @jasonwilliams200OK @mmitche @joshfree

**Comment by benaadams** [comment]
Observations previously about large temp directories potentially causing a race

**Comment by benaadams** [comment]
Have seen "Could not delete workspace" errors prior to the frequency of failures increasing.

**Comment by mmitche** [comment]
@benaadams Did you actively see large temp dirs recently?  I think Chakra saw a few of these recently on smallish temp dirs.  If we do see large temp dirs, then pinpointing where the temp files are coming from is very doable.

**Comment by benaadams** [comment]
Here's a cannot delete workspace issue http://dotnet-ci.cloudapp.net/job/dotnet_coreclr/job/master/job/x86_ryujit_checked_windows_nt_prtest/2971/consoleFull#-4736149696ee26338-6221-4f7c-8c8f-1d1d5cb4704a

**Comment by benaadams** [comment]
And http://dotnet-ci.cloudapp.net/job/dotnet_coreclr/job/master/job/debug_windows_nt_prtest/4718/consoleFull#206289998279494335-f7bd-47d0-8771-8661e00c2db2
```
C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\BuildCustomizations\masm.targets(50,5): 
error MSB4175: The task factory "XamlTaskFactory" could not be loaded from the assembly 
"Microsoft.Build.Tasks.v4.0, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a". 
Could not find file 'C:\Users\dotnet-bot\AppData\Local\Temp\kkocczud.dll'. 
[D:\j\workspace\debug_windows---17180f1b\bin\obj\Windows_NT.x64.Debug\src\debug\di\cordbdi.vcxproj]
```

**Comment by benaadams** [comment]
Maybe the workspace gets deleted early and isn't available for later jobs?

**Comment by jasonwilliams200OK** [comment]
> Maybe the workspace gets deleted early and isn't available for later jobs?

Yeah it seems like Jenkin's environment specific issue. Can't reproduce it locally.

**Comment by mmitche** [comment]
The workspace gets deleted before and after the job, on purpose (disk space).  So it's clear that there was  a file handle that was being held onto in this case.  What's odd is that Jenkins will kill all the child processes launched by the job at the end (to the best of its ability.  Because of the lack of child process information on Windows in some cases, it also uses injected environment variable cookies to locate child processes).

I don't think the two errors are related.

**Comment by jashook** [comment]
@mmitche this has been seen in internal testing as well. I restarted the machines and cleaned out some of the temp storage and have not seen this since.

/cc @RussKeldorph

**Comment by mmitche** [comment]
@jashook we do that also (nightly) and the machines frequently get deleted.  I am thinking that the number of jobs running through each machine is exacerbating the issue.

**Comment by jashook** [comment]
@mmitche that sounds correct to me. Do we have any logging to monitor disk usage over these machines?

**Comment by mmitche** [comment]
@jashook There isn't temp dir logging, though there is logging for overall disk space, Specifically in the past it was the number of temp files in the directory that caused the problem (though the total size was small).

**Comment by mmitche** [comment]
This is also being tracked with: https://github.com/dotnet/dotnet-ci/issues/402 since it is being seen across multiple projects.

**Comment by mmitche** [comment]
I reproduced this outside of Jenkins (on just a regular VM) using ChakraCore, so the Jenkins environment doesn't seem to be the cause.  I'm thinking there may have been some kind of background change in VM perhaps (since this is the same image we have used for 5 months.

**Comment by benaadams** [comment]
Don't know is this is a thing anymore; been a while...

**Comment by mmitche** [comment]
It's not.  Close.

Sent from my iPhone

On Dec 17, 2016, at 5:35 PM, Ben Adams <notifications@github.com<mailto:notifications@github.com>> wrote:


Closed #6746<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fdotnet%2Fcoreclr%2Fissues%2F6746&data=02%7C01%7Cmmitche%40microsoft.com%7C8317ed57c6fe4e3281b708d426ddca51%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636176181414479909&sdata=LfE2DAwHgp2fz9cfadFZMKmosiViApd5Kq%2B0wTUJjJA%3D&reserved=0>.

-
